### PR TITLE
feat: set resource class for a session launcher

### PR DESCRIPTION
**Pull request by andre-code**
## Description

PR to add a feature for selecting a resource class in a session launcher and prompt users to choose a resource class if they lack access to the default one.

## Type of Change
Feat
it requires a new version of [renku-data-services ](https://github.com/SwissDataScienceCenter/renku-data-services/pull/249)

## Related Tickets & Documents

- Related Issue(s): https://github.com/SwissDataScienceCenter/renku/issues/3694
- Closes: https://github.com/SwissDataScienceCenter/renku-ui/issues/3221

Other documents:
- Pitch definition in Notion https://www.notion.so/renku/Set-resource-pool-class-for-a-session-launcher-2-0-c4794c9b0cd840e7a3ea93abd4fa389a

## Screenshots

resource_class_id in launcher

resource_class_id in launcher no access

## UI Accessibility Checklist

For UI changes, ensure the following:

- [x] Use Semantic HTML
- [x] Keyboard Navigation, issue created for bug not fixed here: https://github.com/SwissDataScienceCenter/renku-ui/issues/3222

- [x] Color contrast
- [ ] Lighthouse screenshot
- [ ] Screen Reader Compatibility _Not included_
- [x] Resizing and Zooming

## Testing
Cases to test:
- Select a default resource class when add/update a global session environment launcher
- Select a default resource class when add/update a custom session environment launcher
- Find out which resource class is assigned to a session launcher.
- Display a modal to select a resource class when launching a session and not having access to the default session launcher resource class.
- Be able to select a resource class other than the default when launching a session.

### Added/Updated Unit Tests?

- [ ] Yes
- [x] No

### Added/Updated E2E Tests?

- [X] Yes
- [ ] No

### Renku Acceptance Tests 

Does this involve adding or modifying acceptance tests in [Renku Repository](https://github.com/SwissDataScienceCenter/renku-ui/tree/main/tests)? No

## Storybook
Not included new stories because the components may change due to ongoing app redesigns.

/deploy

### CODE REVIEW

File: client/src/features/session/components/options/SessionClassOption.tsx [hunk 1]
```diff
@@ -58,8 +57,9 @@ import { useCoreSupport } from "../../../project/useProjectCoreSupport";
 import { setSessionClass } from "../../startSessionOptionsSlice";
 import { computeStorageSizes } from "../../utils/sessionOptions.utils";
 
-import styles from "./SessionClassOption.module.scss";
 import { toHumanDuration } from "../../../../utils/helpers/DurationUtils";
+import { FetchingResourcePools } from "../../../sessionsV2/components/SessionModals/ResourceClassWarning.tsx";
```
Comment (leafty) [inline client/src/features/session/components/options/SessionClassOption.tsx:61]:
```suggestion
import { FetchingResourcePools } from "../../../sessionsV2/components/SessionModals/ResourceClassWarning";
```

File: client/src/features/sessionsV2/SessionStartPage.tsx [hunk 2]
```diff
@@ -257,72 +258,36 @@ function StartSessionFromLauncher({
...
   project,
 }: StartSessionFromLauncherProps) {
-  const {
-    containerImage,
-    defaultSessionClass,
-    isFetchingOrLoadingStorages,
-    resourcePools,
-    startSessionOptionsV2,
-  } = useSessionLauncherState({
+  const hasCustomQuery = new URLSearchParams(location.search).has("custom");
```
Comment (leafty) [inline client/src/features/sessionsV2/SessionStartPage.tsx:261]: `"custom"` should probably be defined as a constant. Also, this should use a hook to get the query params:

```suggestion
  const [searchParams] = useSearchParams(); // from "react-router-dom-v5-compat"
  const hasCustomQuery = searchParams.has("custom");
```

File: client/src/features/sessionsV2/components/SessionModals/ModifyResourcesLauncher.tsx
Comment (leafty) [inline client/src/features/sessionsV2/components/SessionModals/ModifyResourcesLauncher.tsx:1]: Note: this form doesn't seem to let users unset the resource class.

**Review by leafty — COMMENTED** [review]
Some code comments:

**Review by leafty — COMMENTED** [review]
Non-blocking: it seems that the project page uses different icons to indicate "edit".

Screenshot 2024-06-27 at 10 27 22
Screenshot 2024-06-27 at 10 27 27

Only one icon should be used for all these cases.

Also the edit button doesn't feel like it's a button at all, there should be some hover and activation styles to communicate to the user that it is an interactive element in the page.

**Comment by andre-code** [comment]
Thanks @leafty for your suggestions. You can find them implemented in the latest commit. Regarding the icon, I share your opinion, but I will create a separate PR for those changes.

File: client/src/features/session/components/options/SessionClassOption.tsx [hunk 3]
```diff
@@ -23,31 +23,30 @@ import {
...
   GroupBase,
+  GroupHeadingProps,
   OptionProps,
   SelectComponentsConfig,
   SingleValue,
   SingleValueProps,
-  GroupHeadingProps,
   components,
 } from "react-select";
+import { useGetNotebooksVersionQuery } from "../../../versions/versions.api.ts";
```
Comment (leafty) [inline client/src/features/session/components/options/SessionClassOption.tsx:38]:
```suggestion
import { useGetNotebooksVersionQuery } from "../../../versions/versions.api";
```

**Review by leafty — APPROVED** [review]

**Pull request merged**